### DISCUSSION TITLE
Investigating the use of a service worker

### DISCUSSION THREAD

**Discussion by erikyo** [WordPress/wp-feature-notifications#247] [REST Controller]
I wanted to propose using a service worker to filter/intercept all requests to the api, that way notifications could even work offline or send system messages

https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API

https://developer.chrome.com/docs/workbox/service-worker-overview/

**Comment by johnhooks** [comment]
Is this just for testing, or for production?

The problem I see with doing that for the end product is figuring out how that fits into the general WordPress system. It's doesn't function offline, so why should the notification system?

**Reply by erikyo** [reply]
For the production, because we can use it to cache // store (and then hydrate) the api responses (however depends on the strategy we want to adopt). 

And yes, it's true that Wordpress doesn't work without a connection but the service worker could just alert about the missing connection to the api while keeping previous responses and/or the notification system active during an intermittent connection (e.g. mobile).

This would be just one of the benefits of using a sw, but for first I think to the performance benefits that we can get using it, and second to system notices that it would be very nice to have them.

**Reply by johnhooks** [reply]
I'm interested to learn service workers and understand the benefits. I'm just curious about how the community will feel about it.

**Reply by erikyo** [reply]
I wrote it here so as to see if there is consensus on this! I hope so, it's not a priority but it would be an interesting stuff for this project.
